mm2src/coins/lightning/ln_events.rs

```diff
@@ -222,28 +299,114 @@ impl LightningEventHandler {
+                let amt_msat = payment_info.amt_msat;
+                if let Err(e) = persister
+                    .add_or_update_payment_in_sql(payment_info)
+                    .await
+                    .error_log_passthrough()
+                {
+                    log::error!("{}", e);
+                }
+                log::info!(
+                    "Successfully sent payment of {} millisatoshis with payment hash {}",
+                    amt_msat.unwrap_or_default(),
+                    hex::encode(payment_hash.0)
+                );
+            }
+        });
+    }
+
+    fn handle_channel_closed(&self, channel_id: [u8; 32], user_channel_id: u64, reason: String) {
+        log::info!(
+            "Channel: {} closed for the following reason: {}",
+            hex::encode(channel_id),
+            reason
+        );
+        let persister = self.persister.clone();
+        let platform = self.platform.clone();
+        // Todo: Handle inbound channels closure case after updating to latest version of rust-lightning
+        // as it has a new OpenChannelRequest event where we can give an inbound channel a user_channel_id
+        // other than 0 in sql
+        if user_channel_id != 0 {
+            spawn(async move {
+                persister
+                    .update_channel_to_closed(user_channel_id, reason)
+                    .await
+                    .error_log();
+                if let Ok(Some(channel_details)) = persister
+                    .get_channel_from_sql(user_channel_id)
+                    .await
+                    .error_log_passthrough()
+                {
+                    if let Some(tx_id) = channel_details.funding_tx {
+                        if let Ok(tx_hash) = H256Json::from_str(&tx_id).error_log_passthrough() {
+                            if let Ok(funding_tx_bytes) = platform
+                                .coin
+                                .as_ref()
+                                .rpc_client
+                                .get_transaction_bytes(&tx_hash)
+                                .compat()
+                                .await
+                                .error_log_passthrough()
+                            {
+                                if let Ok(TransactionEnum::UtxoTx(closing_tx)) = platform
+                                    .coin
+                                    .wait_for_tx_spend(
+                                        &funding_tx_bytes.into_vec(),
+                                        (now_ms() / 1000) + 3600,
+                                        channel_details.funding_generated_in_block.unwrap_or_default(),
+                                        &None,
+                                    )
+                                    .compat()
+                                    .await
+                                    .error_log_passthrough()
+                                {
```

Comment:
This code is very hard to read and the cognitive complexity is probably to high.

Reply:
Done. One question for @artemii235, I don't like error logging when some of these RPC calls fail https://github.com/KomodoPlatform/atomicDEX-API/blob/e680cd3931ad2369ee61956d48e403a02ee3418c/mm2src/coins/lightning/ln_events.rs#L186-L204
I have a lot of these RPC calls in other places too. The reason I don't like this is that if electrums are down, some of the details will be missing from the database due to these RPC calls failing. Do you think looping until the RPC returns a result is a good solution for this? It will solve the problem in all the cases except when mm2 is restarted while electrums are down. what do you think @artemii235?

Reply:
> I have a lot of these RPC calls in other places too

Actually, only this `get_transaction_bytes` call needs retrying. Fixed it on the latest commit along with the restarting case.
For other calls to electrums there are other fail-safes in place.